File: selectionSort.cpp
class Solution
{
    public:
    int select(int arr[], int i)
    {
        // code here such that selectionSort() sorts arr[]
        return 0;
    }
     
    void selectionSort(int arr[], int n)
    {
       //code here
       int min = 0;
       
       for(int i = 0; i < n; i++)
       {
           min = i;
           
           for(int j = i + 1; j < n; j++)
           {
               if(arr[min] > arr[j])
                    min = j;
           }
           
           swap(arr[min], arr[i]);
       }
       
    }
};

File: progress_sheet/selectionSort.py
class Solution: 
    def select(self, arr, i):
        
        min_index = i

        for j in range(i + 1, n):
            
            if arr[min_index] > arr[j]:
                min_index = j
        
        return min_index
    
    def selectionSort(self, arr,n):
        
        for i in range(n):
            
            min_index = self.select(arr, i)
            
            arr[min_index], arr[i] = arr[i], arr[min_index]